### PR TITLE
Exiv2 support old API

### PR DESCRIPTION
**Pull request by fuzzard**
## Description
Cleanup/fix cmake mistakes.
Add support for old (<0.28) cmake config for exiv2
Add support for old (<0.27) libexiv2 API

PRing this so those that care can bikeshed how to deal with the old API. This is no doubt a nasty way to deal with this, but its a start at least i guess.
More than happy for someone else to take this on and make cleaner.

## Motivation and context
Ubuntu users whinging.

## How has this been tested?
Tests pass on Ubuntu 22.04

## What is the effect on users?
Less ubuntu users whinging

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/pictures/metadata/ImageMetadataParser.cpp [hunk 1]
```diff
@@ -75,27 +83,28 @@ std::unique_ptr<ImageMetadata> CImageMetadataParser::ExtractMetadata(const std::
...
-  m_imageMetadata->height = image->pixelHeight();
-  m_imageMetadata->width = image->pixelWidth();
-  m_imageMetadata->fileComment = image->comment();
-
-  if (image->imageType() == Exiv2::ImageType::jpeg)
+  m_imageMetadata->height = image.pixelHeight();
+  m_imageMetadata->width = image.pixelWidth();
+  m_imageMetadata->fileComment = image.comment();
+#if !defined(OLD_EXIV_API)
+  if (image.imageType() == Exiv2::ImageType::jpeg)
```
Comment (enen92) [inline xbmc/pictures/metadata/ImageMetadataParser.cpp:101]: This stuff here requires exactly 0.28.2 or above not just 0.28.0

File: xbmc/pictures/metadata/ImageMetadataParser.cpp [hunk 2]
```diff
@@ -75,27 +83,28 @@ std::unique_ptr<ImageMetadata> CImageMetadataParser::ExtractMetadata(const std::
...
   parser.ExtractIPTC(image->iptcData());
 
   return std::move(parser.m_imageMetadata);
 }
 
-void CImageMetadataParser::ExtractCommonMetadata(std::unique_ptr<Exiv2::Image>& image)
+void CImageMetadataParser::ExtractCommonMetadata(Exiv2::Image& image)
 {
   //! TODO: all these elements are generic should be moved out of the exif struct
-  m_imageMetadata->height = image->pixelHeight();
```
Comment (enen92) [inline xbmc/pictures/metadata/ImageMetadataParser.cpp:89]: TODO above is a leftover, I can clean it later. Members were already moved out of the exif struct

File: xbmc/pictures/metadata/ImageMetadataParser.cpp (hunk 1 above)
Comment (fuzzard) [reply xbmc/pictures/metadata/ImageMetadataParser.cpp:101]: Done.

**Comment by fuzzard** [comment]
@enen92 this includes the debug fix for windows from your other PR (but working)
Ive also added a patch to add a postfix for windows lib name. It would probably be nice to upstream it if you ever find time/desire. I targeted it at only windows lib name to limit the scope of the change.

**Review by enen92 — APPROVED** [review]
I think the way this is handled is easier to nuke once the minimum required version is bumped.
Thanks for cleaning up the mess :)

**Review by thexai — APPROVED** [review]
Tested on Windows - Debug mode

File: xbmc/pictures/metadata/ImageMetadataParser.cpp [hunk 3]
```diff
@@ -16,6 +16,14 @@
 #include <limits>
 #include <vector>
 
+#if (EXIV2_MAJOR_VERSION == 0) && (EXIV2_MINOR_VERSION < 28)
+#define EXIV_UINT32 toLong()
```
Comment (Rechi) [inline xbmc/pictures/metadata/ImageMetadataParser.cpp:20]: I would remove the braces from the defines, so the actual function calls look like function calls. Currently those look like accessing member variables.
Maybe also name the defines `EXIV_toUint32` and `EXIV_toInt64` to be more consistent with the current API names.

**Review by Rechi — APPROVED** [review]

**Pull request merged**